source/extensions/filters/http/ext_proc/mutation_utils.cc

```diff
@@ -24,13 +24,22 @@ using Stats::Counter;
 using envoy::service::ext_proc::v3::BodyMutation;
 using envoy::service::ext_proc::v3::HeaderMutation;
 
-void MutationUtils::headersToProto(const Http::HeaderMap& headers_in,
-                                   envoy::config::core::v3::HeaderMap& proto_out) {
-  headers_in.iterate([&proto_out](const Http::HeaderEntry& e) -> Http::HeaderMap::Iterate {
-    auto* new_header = proto_out.add_headers();
-    new_header->set_key(std::string(e.key().getStringView()));
-    new_header->set_value(MessageUtil::sanitizeUtf8String(e.value().getStringView()));
+bool MutationUtils::headerInAllowList(
```

Comment:
For the case that multiple ext_proc filters in the chain with same forwarding rule, we probably can cache and share the match result.

But this is out of scope of this PR and it's only safe when forwarding rule is same (i.e., with indication from config). We can think about/discuss this later.

Reply:
Yeah, we can discuss that implementation optimization later.

Reply:
> Thank you for working on this!
> 
> Mostly LGTM, just a few comments.

Thanks for the comments!